client/my-sites/patterns/components/get-access-modal/index.tsx

```diff
@@ -1,22 +1,50 @@
 import { Button, Dialog } from '@automattic/components';
 import { useLocalizeUrl, useLocale } from '@automattic/i18n-utils';
 import { Icon, close as iconClose } from '@wordpress/icons';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { getTracksPatternType } from '../../lib/get-tracks-pattern-type';
+import { PatternTypeFilter } from '../../types';
 
 import './style.scss';
 
 type PatternsGetAccessModalProps = {
+	category: string;
 	isOpen: boolean;
 	onClose: () => void;
+	pattern: string;
+	patternTypeFilter: PatternTypeFilter;
 };
 
-export const PatternsGetAccessModal = ( { isOpen, onClose }: PatternsGetAccessModalProps ) => {
+export const PatternsGetAccessModal = ( {
+	category,
+	isOpen,
+	onClose,
+	pattern,
+	patternTypeFilter,
+}: PatternsGetAccessModalProps ) => {
 	const locale = useLocale();
 	const localizeUrl = useLocalizeUrl();
 
 	const isLoggedIn = false;
 	const startUrl = localizeUrl( '//wordpress.com/start/account/user', locale, isLoggedIn );
 	const loginUrl = localizeUrl( '//wordpress.com/log-in', locale, isLoggedIn );
 
+	const recordSignupClickEvent = ( tracksEventName: string ) => {
+		recordTracksEvent( tracksEventName, {
+			name: pattern,
+			category,
+			type: getTracksPatternType( patternTypeFilter ),
+		} );
+	};
+
+	const recordLoginClickEvent = ( tracksEventName: string ) => {
+		recordTracksEvent( tracksEventName, {
+			name: pattern,
+			category,
+			type: getTracksPatternType( patternTypeFilter ),
+		} );
+	};
```

Comment:
We have totally the same handlers. So we should:
1) Either - make just 1 handler which receives property `tracksEventName`
2) Or - different handlers w/o arguments, but with encapsulated `tracksEventName`

I don't have a preference for 1st or 2nd, so up to you which you like more

Reply:
Oh, moreover - we have teh same argument as `calypso_pattern_library_get_access_signup`. I suppose it should be another for `Log in`, right? Is it typo?

Reply:
oof, yes. definitely a typo. These events were so similar it looks like I fell into the copypasta trap and missed that modification. Will fix that typo now while I look at streamlining these repetitive handlers a bit!